### DISCUSSION TITLE
🔗 Posibility to switch OFF Elementor AI (Related to #22796)

### DISCUSSION THREAD

**Discussion by ghost** [elementor/elementor#22853] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I dont need this feature, cause working only at english language, and paid. I dont want to see this feature, better will be if in setting we will have ability to shutdown this AI. 

### Describe the solution you'd like

In settings need button (or radio) to shutdown / enable AI.

**Comment by lensbreak** [comment]
+1 but pretty sure they are not gonna do that to push their new part of business model (see subscription for AI credits). also still waiting them to fix the role manager so containers design can not be changed by others, only content acces. it will takes another 2 years as always i guess...

**Reply by ghost** [reply]
Maybe... 

But if this are open source (GPL) and we can disable / enable features like containers / others ... AI this is feature too. And if i DONT want use it - i must have posibility to disable this)) 

But at default - can be enabled. This is not a problem)

**Comment by ronnyadsetts** [comment]
I arrived here looking for the same thing. After editing one of our customer's sites, I was frustrated by the 'rewrite with AI' links absolutely everywhere! I have no desire nor need for this "feature", neither do I want our non-technical website customers to be faced with this! We have to support this nonsense. I appreciate that some folks may want to have computers write their content but please make it optional in the UI.

I mean, this is on the Pro plugin where you don't expect up-sells. We're paying already, thank you!

**Comment by iMiek** [comment]
Totally agree with @ronnyadsetts. 
I own sites with photo galleries and it's annoying. All images are check by "AI" for alt-descriptions. It's slowing down a lot!!! Pleas add the possibility to switch AI off.

**Comment by iMiek** [comment]
Found a tip on wppagebuilders.com → disable-ai-elementor:
You can use your theme functions.php file which you can access from the Appearance → Theme File Editor or other means (such as using the Code Snippet plugin) that allow you to add_action to your WordPress website. 

Below is the code snippet to disable AI in Elementor:
```
add_action('elementor/editor/before_enqueue_scripts', function () {
    echo "<style>";
    echo ".e-ai-button {display: none;}";
    echo "</style>";
  
});
```

**Reply by ronnyadsetts** [reply]
Of course, that only hides the AI button in the UI. Anything that's happening in the background with Elementor and AI will still happen. I guess it would however prevent our customers from clicking it in their naivety.

**Reply by iMiek** [reply]
I know. But it's a temporary solution till Elementor gives a better one.....

**Comment by perks98** [comment]
Not sure if this is useful to anyone or if this works well but if you use the following PHP in a child theme it should remove the AI button on controls so its HTML doesn't render rather than using CSS to hide, just an alternative.

```
function remove_ai_control( $element, $section_id, $args ) {

		foreach( $element->get_controls() as $control ) {
			$element->update_control(
				$control['name'],
				[
					'ai' => false,
				]
			);
		}
}
add_action( 'elementor/element/before_section_start', 'remove_ai_control', 10, 3 );
```

**Reply by squadcar** [reply]
Seems to do the trick! Thanks @perks98 :)

**Comment by c2wd** [comment]
```
function remove_elementor_ai_script_from_editor() {
    if ( \Elementor\Plugin::$instance->editor->is_edit_mode() ) {
        wp_dequeue_script( 'elementor-ai' );
    }
}
add_action( 'elementor/editor/after_enqueue_scripts', 'remove_elementor_ai_script_from_editor' );
```

**Reply by squadcar** [reply]
This, combined with the one above, seems to kill 99.99% of the AI junk. Haven't seen any controls or popups anywhere since.

**Comment by kristiangl** [comment]
100% agree. We should be able to switch this off (also switch off the Elementor Optimizer prompts) that are everywhere, especially for client websites. It adds so much visual clutter and makes it so much harder/more intimidating for clients who aren't as technologically minded.

**Comment by speedxcz** [comment]
I agree, I get 5 times a day from customers angry emails why they have an ad on a paid site. We pay for a plugin and don't want to constantly see advertising on Ai, or on a plugin to optimize images etc... Pretty soon it's going to be that we have to click through a forest of ads to get to the site edits.

**Comment by mbaierl** [comment]
@see https://github.com/orgs/elementor/discussions/27354 - my bug request has been closed. 
Seems there is no awareness for how agencies work and what troubles those upsells cause.

**Comment by nicholaszein** [comment]
This Discussion was referenced in the following issues and discussions:

**Discussions**
- #22796

**Comment by nicholaszein** [comment]
_A word of advice, and a warning: Offensive or Disrespectful remarks, or any behavior that does not align with the [Code of Conduct](https://elemn.to/gh-code-of-conduct) will grant the author a block from contributing or participating in this repository. Please, be respectful. Keep the conversation amicable and civilized. I have no pleasure in having to block users. I am a user myself. But I will do it if I see any comment breaking the Code of Conduct._

**Reply by ghost** [reply]
I dont seen any disrespectful remarks. 

WE just want to be listened and developer must do some buttons to enable / disable functions which u dont need.
Cause this is ruin our business / job. 

Can you speak with them and say our opinion? will be great.

WE DONT want to delete this function. But we want just to have posibility to disable it.

**Reply by squadcar** [reply]
No kidding, people expressing their frustration with an issue that has been affecting pro users for at least a year now, with zero response from Elementor, is not being disrespectful. Ignoring your pro users on the other hand...

**Reply by nicholaszein** [reply]
@necrosoul and @squadcar you don't see any offensive comments because I had to delete them. 🫤

**Comment by ghost** [comment]
I think developers does not care about us. 
We see updates at AI, but NO changes for this situation.

@rami-elementor @ramiy  if we NOT using AI, we will NOT use this at future. 
NO needed to force us to buy this feature. 

I understand this is business but ... you will not get any money for AI from ppl who hate this function.

**Reply by ghost** [reply]
@arielk i hope you will change it.

**Comment by ronnyadsetts** [comment]
We now have Elementor AI imposing itself in the WP Media Library. The element IDs to "display: none" are currently:

#e-image-ai-media-library
#e-image-ai-attachment-details
#e-image-ai-insert-media

You can also (or instead) dequeue the script "elementor-ai-media-library".

This is turning in to some sort of customer versus vendor war. Whatever happened to the customer always being right?

**Reply by ghost** [reply]
This situation not only in Elementor, this is in game dev / others. 
Vendor just does not care about customer. 

We had addon "Flexible elementor panel" ( i think this is true name ) and i wonder WHY this is not default function in Elementor. 
I seen addon "Designer powerup" and again.. dont understand why this is NOT in Elementor. Too much good functions. 

But i see AI.. and spam. Why? Dont know. 

I will not be surprised if rating of Elementor will be decreased and customers will go to other builders, or will see in wordpress plugins "Elementor shutdown trash" where will be disabled all AI scam / other things.

**Comment by c2wd** [comment]
Since Elementor has pushed more Elementor AI on us instead of listening and giving us the option of turning this garbage off, I have updated some snippets to help get rid of it for those that don't want it imposing on their workflow.

**Remove Elementor AI from the Editor:**
```
function remove_elementor_ai_script_from_editor() {
    if ( \Elementor\Plugin::$instance->editor->is_edit_mode() ) {
        wp_dequeue_script( 'elementor-ai' );
        wp_dequeue_script( 'elementor-ai-media-library' );
    }
}
add_action( 'elementor/editor/after_enqueue_scripts', 'remove_elementor_ai_script_from_editor' );
```

**Remove Elementor AI from the Media Library:**
```
function dequeue_elementor_ai_media_library_admin() {
    wp_dequeue_script('elementor-ai-media-library');
}
add_action('admin_enqueue_scripts', 'dequeue_elementor_ai_media_library_admin', 100);
```

**Reply by ghost** [reply]
helpful) sad that devs does not care about us

**Reply by Mrsnibs** [reply]
Thanks for this. Works when combined with perks98's html solution for the buttons. What a ridiculous situation for an important plugin.

**Reply by reardenDev** [reply]
I owe you one

**Comment by ghost** [comment]
```
3.23.1 – 2024-07-15

Fix: Slow loading of WordPress admin and Elementor screens due to AI health check
```

How about start listening to us? How about to make posibility to switch off AI? 
Why you hate your clients?

**Comment by ghost** [comment]
I was directed to a 'Feature Request' area of all places (here), to "complain" (those are Elementor's words...see below) because of the BUG that is Elementor's AI.

**It's a BUG because it is breaking our sites, slowing them down, causing issues with clients, and the list goes on.**

_"Hello @wpturnedup.

Please do not use GitHub Issues to file complaints or request features. Elementor GitHub Issues are reserved only to report bugs in the code. The feature you're referring to is not a bug.

If you need to file a complaint reach out to Elementor Support. If you wish to request a feature check our [Feature Request Discussions](https://elemn.to/gh-feature-requests).

Keep in mind, that by using this space you agree to our [Contribution Guidelines](https://elemn.to/gh-contributing) and [Code of Conduct](https://elemn.to/gh-code-of-conduct).

Here is a Feature Request related to your issue that you can join and vote: https://github.com/orgs/elementor/discussions/22853

Kind regards"_

**Comment by c2wd** [comment]
Here is an updated code snippet to hopefully combat the latest round of AI and Image Optimization NAGS from Elementor. Feel free to give input and/or feedback so everyone can implement and get back to a nice clean workflow for ourselves and our clients.

```
function customize_elementor() {
    // Remove Elementor Nag Admin Notices
    echo '<style>.e-notice--cta { display: none !important; }</style>';

    // Remove Elementor AI script from editor
    if ( \Elementor\Plugin::$instance->editor->is_edit_mode() ) {
        wp_dequeue_script( 'elementor-ai' );
        wp_dequeue_script( 'elementor-ai-media-library' );
    }

    // Dequeue Elementor AI media library and Gutenberg scripts in admin
    wp_dequeue_script( 'elementor-ai-media-library' );
    wp_dequeue_script( 'elementor-ai-gutenberg' );
}

// Hook for removing Elementor nag in admin
add_action( 'admin_head', 'customize_elementor' );

// Hook for removing Elementor AI scripts from editor
add_action( 'elementor/editor/after_enqueue_scripts', 'customize_elementor' );

// Hook for dequeuing Elementor AI media library and Gutenberg scripts in admin
add_action('admin_enqueue_scripts', 'customize_elementor', 100);

```

**Reply by ghost** [reply]
thanks. will test it.

**Comment by mikeecho2** [comment]
And basically guaranteed that this will remain an issue for YEARS TO COME, as Elementor is more interested in being AI & HOSTING COMPANY than a Page Builder Company, like the USE TO BE

**Reply by ghost** [reply]
Sad but true. Only money they need.

**Reply by ghost** [reply]
I’ve been one of the biggest “complainers“  regarding this topic. I did a live stream with The WP Minute and the head of WP Relations at Elementor. It seems that changes (for the good) are coming soon. I’m hopeful, but we’ll have to wait and see.

The biggest thing I took away from all this is that to facilitate change, it requires more than just words in a GitHub account. You have to demonstrate (videos were my medium) and discuss the “pain” outside of GitHub. Think blog, YouTube, etc., and in doing so come at it from an angle of caring about the tool you use rather than tearing it down. 

Now, the issue of how this whole GitHub interaction has been (hasn’t been) working for us is still flawed IMO. Decisions made here in GitHub, should be more openly made with more Elementor folks involved. Seemingly, one person is the gatekeeper at the moment and it’s not working.

**Reply by ghost** [reply]
> I’ve been one of the biggest “complainers“ regarding this topic. I did a live stream with The WP Minute and the head of WP Relations at Elementor. It seems that changes (for the good) are coming soon. I’m hopeful, but we’ll have to wait and see.
> 
> The biggest thing I took away from all this is that to facilitate change, it requires more than just words in a GitHub account. You have to demonstrate (videos were my medium) and discuss the “pain” outside of GitHub. Think blog, YouTube, etc., and in doing so come at it from an angle of caring about the tool you use rather than tearing it down.
> 
> Now, the issue of how this whole GitHub interaction has been (hasn’t been) working for us is still flawed IMO. Decisions made here in GitHub, should be more openly made with more Elementor folks involved. Seemingly, one person is the gatekeeper at the moment and it’s not working.

Yes., problem is.. they not read us at Github, and at wordpress repository too. I cant create video. And dont know big bloggers who will speak about this.

**Comment by miriamelementor** [comment]
I just wanted to share an update: that we have removed the majority of AI popups. We are also working on a way for users to easily disconnect their sites from AI. This functionality is in progress and we will update here as soon as it’s released.

**Reply by ghost** [reply]
Hope that will be removed not MAJORITY, but ALL AI FEATURES.  DONT want to see this in my fav. plugin.

**Reply by ghost** [reply]
> I just wanted to share an update: that we have removed the majority of AI popups. We are also working on a way for users to easily disconnect their sites from AI. This functionality is in progress and we will update here as soon as it’s released.

And please make ALL Features like AI / image optimization and other "scam" functions posible to switch off completely. 
NOT delete. NOT disabled by default.. but make posible to disable in settings.  This will be good for all.

**Comment by toineenzo** [comment]
FYI ;) https://github.com/elementor/elementor/pull/28303
Something's in the works

**Reply by toineenzo** [reply]
And it has been committed https://github.com/elementor/elementor/commit/ef52a9919ae20938856707e1171da38b8e50a7b4 !!

**Comment by ghost** [comment]
Good. they listening to us.. but why disable per user? need sitewide disable too.

**Comment by clmnsm** [comment]
It doesn't seem to be hookable, maybe deliberately so. Disappointing.
